provers/groth16/src/prover.rs

```diff
@@ -0,0 +1,147 @@
+use crate::{common::*, ProvingKey, QAP};
+use lambdaworks_math::errors::DeserializationError;
+use lambdaworks_math::traits::{Deserializable, Serializable};
+use lambdaworks_math::{cyclic_group::IsGroup, msm::pippenger::msm};
+use std::mem::size_of;
+
+pub struct Proof {
+    pub pi1: G1Point,
+    pub pi2: G2Point,
+    pub pi3: G1Point,
+}
+
+impl Proof {
+    pub fn new(w: &[FrElement], qap: &QAP, pk: &ProvingKey) -> Self {
```

Comment:
I think it would be better to have a `Prover` struct with a method `prove` that generates the proof. That is closer to what one finds in the literature and would be easier for newcommers to understand the code.

```suggestion
pub struct Proof {
    pub pi1: G1Point,
    pub pi2: G2Point,
    pub pi3: G1Point,
}
struct Prover;

impl Prover {
    pub fn prove(w: &[FrElement], qap: &QAP, pk: &ProvingKey) -> Proof {
```

Reply:
ok

Reply:
@irfanbozkurt I'm also working on this, let's sync on this if possible

Reply:
implemented this